src/cdk/schematics/update-tool/utils/imports.ts

```diff
@@ -8,39 +8,106 @@
 
 import * as ts from 'typescript';
 
-export type Import = {
-  name: string,
-  importModule: string,
-  node: ts.ImportDeclaration
-};
-
-/** Gets import information about the specified identifier by using the type checker. */
-export function getImportOfIdentifier(typeChecker: ts.TypeChecker, node: ts.Identifier): Import|
+/** Interface describing a resolved import. */
+export interface Import {
+  /** Name of the imported symbol. */
+  symbolName: string;
+  /** Module name from which the symbol has been imported. */
+  moduleName: string;
+}
+
+
+/** Resolves the import of the specified identifier. */
+export function getImportOfIdentifier(node: ts.Identifier, typeChecker: ts.TypeChecker): Import|
     null {
-  const symbol = typeChecker.getSymbolAtLocation(node);
+  const directImport = getSpecificImportOfIdentifier(node, typeChecker);
+  if (directImport !== null) {
+    return directImport;
+  } else if (ts.isQualifiedName(node.parent) && node.parent.right === node) {
+    const qualifierRoot = getQualifiedNameRoot(node.parent);
+    if (qualifierRoot) {
+      const moduleName = getImportOfNamespacedIdentifier(qualifierRoot, typeChecker);
+      if (moduleName) {
+        return {moduleName, symbolName: node.text};
+      }
+    }
+  } else if (ts.isPropertyAccessExpression(node.parent) && node.parent.name === node) {
+    const rootIdentifier = getPropertyAccessRoot(node.parent);
+    if (rootIdentifier) {
+      const moduleName = getImportOfNamespacedIdentifier(rootIdentifier, typeChecker);
+      if (moduleName) {
+        return {moduleName, symbolName: node.text};
+      }
+    }
+  }
+  return null;
+}
 
+/**
+ * Resolves the import of the specified identifier. Expects the identifier to resolve
+ * to a fine-grained import declaration with import specifiers.
+ */
+function getSpecificImportOfIdentifier(node: ts.Identifier, typeChecker: ts.TypeChecker): Import|
+    null {
+  const symbol = typeChecker.getSymbolAtLocation(node);
   if (!symbol || !symbol.declarations || !symbol.declarations.length) {
     return null;
   }
-
-  const decl = symbol.declarations[0];
-
-  if (!ts.isImportSpecifier(decl)) {
+  const declaration = symbol.declarations[0];
+  if (!ts.isImportSpecifier(declaration)) {
     return null;
   }
-
-  // Since "decl" is an import specifier, we can walk up three times to get a reference
-  // to the import declaration node (NamedImports -> ImportClause -> ImportDeclaration).
```

Comment:
Why delete this comment?

Reply:
Not intentional. I just moved the whole file from the v9 hammer migration and this apparently didn't have that comment. Re-added.